Unclosed scandir iterator

I'm seeing the following warning fairly frequently in code that uses local fsspec (via https://github.com/pystatgen/sgkit/runs/2981156269?check_suite_focus=true):

```
E               pytest.PytestUnraisableExceptionWarning: Exception ignored in: <posix.ScandirIterator object at 0x7f869c1281f0>
E               
E               Traceback (most recent call last):
E                 File "/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/fsspec/spec.py", line 394, in walk
E                   return path, {}, {}
E               ResourceWarning: unclosed scandir iterator <posix.ScandirIterator object at 0x7f869c1281f0>
```

I think this can be fixed by closing the scandir iterator in the `ls` method of `LocalFileSystem`.

--- a/fsspec/implementations/local.py
+++ b/fsspec/implementations/local.py
@@ -51,7 +51,8 @@ class LocalFileSystem(AbstractFileSystem):
     def ls(self, path, detail=False, **kwargs):
         path = self._strip_protocol(path)
         if detail:
-            return [self.info(f) for f in os.scandir(path)]
+            with os.scandir(path) as it:
+                return [self.info(f) for f in it]
         else:
             return [posixpath.join(path, f) for f in os.listdir(path)]
 

